docs: modified `01_concept.md` to describe two auction types

jeonghoyeo7 opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

closes: #81 

## Tasks

- Modified `01_concept.md` by adding the description of batch auction type and also by modifying the existing descriptions.

## References


---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

jeonghoyeo7 committed:
first draft for 01_concept

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types #81

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types (#81)

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types (#81)

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types (#81)

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types (#81)

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types (#81)

jaybxyz commented on x/fundraising/spec/01_concepts.md:
```diff
@@ -4,18 +4,71 @@
+When an auctioneer creates a fixed price auction, it must determine the following parameters.
 
-An english auction type is an ascending dynamic auction that an auctioneer decides the starting price of the selling amount of coin and bidders bid to purchase the amounts of the coin. Competition for the price, not how fast bidders can bid.
+- **Selling Token**: the denom of tokens to be auctioned,
+- **Paying Token**: the denom of tokens to be used for payment,
+- **Price**: fixed amount of the paying tokens to get a selling token (i.e., amount of paying tokens per selling token),
+- **Auction Start Time**: when the auction starts,
+- **Auction End Time**: when the auction ends,
+- **Offering Quantity**: total amount of selling tokens to be auctioned.
```
I update names to be the exact names that are used in the codebase and add `Vesting Schedules`. You can reference what parameters that an auctioneer needs to input in [this /docs/How-To/cli/README.md](https://github.com/tendermint/fundraising/tree/main/docs/How-To/cli#createfixedpriceauction).
```suggestion
- **Price**: fixed amount of the paying tokens to get a selling token (i.e., amount of paying tokens per selling token),
- **Selling Coin**: the total amount of selling coin to be auctioned,
- **Paying Coin Denom**: the denom of coin to be used for payment,
- **Vesting Schedules**: the vesting schedules to receive the sold amounts,
- **Auction Start Time**: when the auction starts,
- **Auction End Time**: when the auction ends.
```

jeonghoyeo7 replied:
Good suggestions. I made one more change to add "denom and" to Selling Coin.
```suggestion
- **Price**: fixed amount of the paying tokens to get a selling token (i.e., amount of paying tokens per selling token),
- **Selling Coin**: the denom and total amount of selling coin to be auctioned,
- **Paying Coin Denom**: the denom of coin to be used for payment,
- **Vesting Schedules**: the vesting schedules to receive the sold amounts,
- **Auction Start Time**: when the auction starts,
- **Auction End Time**: when the auction ends.
```

We are now using both "token" and "coin" here and there. What do you think we use one term between two? Which one is better you think?

jaybxyz replied:
We have internally discussed this before. We decided to use `coin` because both selling coin and paying coin are native assets that operate on its own blockchain.

jeonghoyeo7 replied:
Thanks for the info. I will modify accordingly.

jaybxyz commented on x/fundraising/spec/01_concepts.md:
```diff
@@ -4,18 +4,71 @@
+- **Offering Quantity**: total amount of selling tokens to be auctioned.
 
-## Extended Auction Round(s)
+### What bidders can/cannot do:
```
Since we use a singular for an auctioneer above `### What an auctioneer does`, I think it is better to make it consistent. What do you think?
```suggestion
### What a bidder can/cannot do:
```

jeonghoyeo7 replied:
It looks better. Thanks.

jaybxyz commented on x/fundraising/spec/01_concepts.md:
```diff
@@ -4,18 +4,71 @@
+
+### When the auction ends:
+
+The auction will end either when Auction End Time is arrived or when the entire Offering Quantity is sold out.
```
Suggestion 1
```suggestion
The auction will end either when the end time is arrived or when the entire selling coin is sold out.
```

Suggestion 2
```suggestion
The auction will end either when the `EndTime` is arrived or when the entire `SellingCoin` is sold out.
```

jeonghoyeo7 replied:
I'm modifying based on Suggestion 2 since the variable names will be used in many places.

jaybxyz commented on x/fundraising/spec/01_concepts.md:
```diff
@@ -4,18 +4,71 @@
+- **Auction End Time**: when the auction ends,
+- **Offering Quantity**: total amount of selling tokens to be auctioned.
+
+### What bidders can/cannot do:
```
Same comment as above.

jaybxyz commented on x/fundraising/spec/01_concepts.md:
```diff
@@ -4,18 +4,71 @@
+
+When an auctioneer creates this batch auction, it must determine the following parameters.
+
+- **Selling Token**: the denom of tokens to be auctioned,
+- **Paying Token**: the denom of tokens to be used for payment,
+- **Auction Start Time**: when the auction starts,
+- **Auction End Time**: when the auction ends,
+- **Offering Quantity**: total amount of selling tokens to be auctioned.
```
Same comment as above. I suggest to use names to be the exact names that are used in codebase and an auctioneer needs to input `Vesting Schedules`.

jaybxyz reviewed (changes requested):
Great work @jeonghoyeo7 👍  

I left some comments and suggestions below. Please take a look and let me know what you think. And there is `MsgCancelAuction` for an auctioneer to cancel their auction when the auction has not started yet. Do you think it is better to include the content in `01_concepts.md`?

jeonghoyeo7 committed:
Update x/fundraising/spec/01_concepts.md

Co-authored-by: JayB <jaybkim2@gmail.com>

jeonghoyeo7 committed:
Update x/fundraising/spec/01_concepts.md

Since we didn't use variable names in other places, Suggestion 1 looks better.

Co-authored-by: JayB <jaybkim2@gmail.com>

jeonghoyeo7 committed:
Update x/fundraising/spec/01_concepts.md

Co-authored-by: JayB <jaybkim2@gmail.com>

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types (#82)

jeonghoyeo7 commented:
@kogisin I revised based on the comments: mainly reflect the variable names. I think I need more modification afterwards when Batch Auction is being implemented. There is `Coin` in a bid, but for the two different bidding options in batch auction,  `bid` can be a paying coin or a selling coin for bidding. But `PayingCoin` and `SellingCoin` are used in the type of auction, so how to differently describe those seems to be needed. I will further modify those parts after that aspect is resolved.

jaybxyz reviewed (approved):
LGTM

jaybxyz committed:
Merge branch 'main' into jeongho/spec

jaybxyz commented:
I merged the latest commit from `main` branch to this PR so that the broken test is fixed.

jeonghoyeo7 committed:
Update 01_concepts.md

jeonghoyeo7 committed:
docs: add/modify specs to describe auction types

- modify three minor parts

jeonghoyeo7 commented:
- I modified three minor parts (changes by using variables)
  - Please check and, if ok, then we can merge this.
